third_party/py/python_configure.bzl

```diff
@@ -0,0 +1,330 @@
+  result = repository_ctx.execute([_get_bash_bin(repository_ctx), "-c", cmd])
+  return result.stdout.strip('\n')
+
+
+def _check_python_lib(repository_ctx, python_lib):
+  """Checks the python lib path."""
+  cmd = 'test -d "%s" -a -x "%s"' % (python_lib, python_lib)
+  result = repository_ctx.execute([_get_bash_bin(repository_ctx), "-c", cmd])
+  if result.return_code == 1:
+    _fail("Invalid python library path: %s" % python_lib)
+
+
+def _check_python_bin(repository_ctx, python_bin):
+  """Checks the python bin path."""
+  cmd =  '[[ -x "%s" ]] && [[ ! -d "%s" ]]' % (python_bin, python_bin)
+  result = repository_ctx.execute([_get_bash_bin(repository_ctx), "-c", cmd])
+  if result.return_code == 1:
+    _fail("--define %s='%s' is not executable. Is it the python binary?" % (
+        _PYTHON_BIN_PATH, python_bin))
+
+
+def _get_python_include(repository_ctx, python_bin):
+  """Gets the python include path."""
+  result = _execute(
+      repository_ctx,
+      [python_bin, "-c",
+       'from __future__ import print_function;' +
+       'from distutils import sysconfig;' +
+       'print(sysconfig.get_python_inc())'],
+      error_msg="Problem getting python include path.",
+      error_details=("Is the Python binary path set up right? " +
+                     "(See ./configure or " + _PYTHON_BIN_PATH + ".) " +
+                     "Is distutils installed?"))
+  return result.stdout.splitlines()[0]
+
+
+def _get_python_import_lib_name(repository_ctx, python_bin):
+  """Get Python import library name (pythonXY.lib) on Windows."""
+  result = _execute(
+      repository_ctx,
+      [python_bin, "-c",
+       'import sys;' +
+       'print("python" + str(sys.version_info[0]) + ' +
+       '      str(sys.version_info[1]) + ".lib")'],
+      error_msg="Problem getting python import library.",
+      error_details=("Is the Python binary path set up right? " +
+                     "(See ./configure or " + _PYTHON_BIN_PATH + ".) "))
+  return result.stdout.splitlines()[0]
+
+
+def _get_numpy_include(repository_ctx, python_bin):
+  """Gets the numpy include path."""
+  return _execute(repository_ctx,
+                  [python_bin, "-c",
+                   'from __future__ import print_function;' +
+                   'import numpy;' +
+                   ' print(numpy.get_include());'],
+                  error_msg="Problem getting numpy include path.",
+                  error_details="Is numpy installed?").stdout.splitlines()[0]
+
+
+def _create_local_python_repository(repository_ctx):
+  """Creates the repository containing files set up to build with Python."""
+  python_bin = _get_python_bin(repository_ctx)
+  _check_python_bin(repository_ctx, python_bin)
+  python_lib = _get_python_lib(repository_ctx, python_bin)
+  _check_python_lib(repository_ctx, python_lib)
+  python_include = _get_python_include(repository_ctx, python_bin)
+  # numpy_include = _get_numpy_include(repository_ctx, python_bin) + '/numpy'
```

Comment:
We generally don't allow commented-out code in the repository. Should this be some special exception? Should it be a TODO if the commented-out code suggests something to be done in the future?